tests/test_flow/test_persistence_aip.py

```diff
@@ -107,3 +115,67 @@ def x_plus_one(x):
         r"Submitting AI Platform task .*\(name='x_plus_one'\).*CaseKey\(x=1\).*",
         r".*error while doing remote computation for x_plus_one\(x=1\).*AipError.*",
     )
+
+
+def test_fix_pip_requirements():
+    pip_requirements = dedent(
+        """
+        Package1==1.2.3
+        Package2==2
+        -e git+git@github.com:square/bionic.git@f13f5405e928d92b553d2cbee41084eecccf7de3#egg=bionic
+        -e git+https://github.com/square/bionic.git@88fec3d6921ed13b7c7575cca4c292b4f7003b9c#egg=bionic
+    """
+    )
+
+    fixed_pip_requirements = dedent(
+        """
+        Package1==1.2.3
+        Package2==2
+        -e git+https://github.com/square/bionic.git@f13f5405e928d92b553d2cbee41084eecccf7de3#egg=bionic
+        -e git+https://github.com/square/bionic.git@88fec3d6921ed13b7c7575cca4c292b4f7003b9c#egg=bionic
+    """
+    )
+
+    assert fix_pip_requirements(pip_requirements) == fixed_pip_requirements
+
+
+def git_repo_is_clean_and_pushed():
+    repo = Repo(os.getcwd(), search_parent_directories=True)
+    return (
+        not repo.is_dirty()
+        and len(repo.git.branch("-r", "--contains", repo.head.ref.object.hexsha)) > 0
+    )
+
+
+@pytest.mark.skipif(
+    not git_repo_is_clean_and_pushed(),
+    reason="current commit needs to be available in remote repo for docker access",
+)
+@pytest.mark.real_gcp_only
+@pytest.mark.no_parallel
+def test_aip_with_docker_build(aip_builder, gcp_project):
```

Comment:
Is there any way to test just the image building part, or to test it together with a fake AIP? (Not necessarily in this PR -- just curious how hard it would be.)

Reply:
I think it should be possible to do it on github. There are github actions that can install docker on the test environment. My concern is that it would take some amount of time downloading and installing all the python packages and consuming github quota.